src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs

```diff
@@ -840,21 +840,10 @@ private unsafe void ReadItemsFromSource(ref UnmanagedWriteBuffer writeBuffer, in
                     Debug.Assert(collectionSize > 0);
                     item.Collection = Encoding.UTF8.GetString(ReadExactly(collectionSize), collectionSize);
 
-                    var nameSize = *(int*)ReadExactly(sizeof(int));
-                    item.CounterName = Encoding.UTF8.GetString(ReadExactly(nameSize), nameSize);
-
-                    item.CounterValue = *(long*)ReadExactly(sizeof(long));
-                }
-                else if (item.Type == ReplicationBatchItem.ReplicationItemType.CounterTombstone)
-                {
-                    var keySize = *(int*)ReadExactly(sizeof(int));
-                    item.KeyDispose = Slice.From(context.Allocator, ReadExactly(keySize), keySize, out item.Key);
-
-                    var collectionSize = *(int*)ReadExactly(sizeof(int));
-                    Debug.Assert(collectionSize > 0);
-                    item.Collection = Encoding.UTF8.GetString(ReadExactly(collectionSize), collectionSize);
+                    var sizeOfData = *(int*)ReadExactly(sizeof(int));
+                    item.DocumentSize = sizeOfData;
 
-                    item.LastModifiedTicks = *(long*)ReadExactly(sizeof(long));
+                    ReadExactly(sizeOfData, ref writeBuffer);
```

Comment:
Need to handle 4.1 -> 4.2 replication counter (error, abort connection)
4.2 -> 4.1 want to replicate (error, abort connection)

Reply:
done